src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs

```diff
@@ -148,6 +148,17 @@ public async Task Invoke(HttpContext context)
                     context.Response.StatusCode = 500;
                 }
 
+                var exceptionHandlerFeature = new ExceptionHandlerFeature()
+                {
+                    Error = ex,
+                    Path = context.Request.Path,
+                    Endpoint = context.GetEndpoint(),
+                    RouteValues = context.Features.Get<IRouteValuesFeature>()?.RouteValues
+                };
+
+                context.Features.Set<IExceptionHandlerFeature>(exceptionHandlerFeature);
+                context.Features.Set<IExceptionHandlerPathFeature>(exceptionHandlerFeature);
+
                 await _exceptionHandler(new ErrorContext(context, ex));
```

Comment:
nit: Move `new ErrorContext(...)` into a local and call `SetExceptionHandlerFeatures`

Reply:
There is a `SetExceptionHandlerFeatures` method already.
https://github.com/dotnet/aspnetcore/blob/1a3fda55135cfdb0f81da8c51e3272902103ea4b/src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs#L84
You can call it if you just move `new ErrorContext` into a local.
```
var ec = new ErrorContext(context, ex);
SetExceptionHandlerFeatures(ex);
await _exceptionHandler(ec);
```